themes/bureau.zsh-theme

```diff
@@ -23,44 +23,51 @@ bureau_git_branch () {
 }
 
 bureau_git_status() {
-  _STATUS=""
+  local gstatus
+  gstatus=""
 
   # check status of files
-  _INDEX=$(command git status --porcelain 2> /dev/null)
-  if [[ -n "$_INDEX" ]]; then
-    if $(echo "$_INDEX" | command grep -q '^[AMRD]. '); then
-      _STATUS="$_STATUS$ZSH_THEME_GIT_PROMPT_STAGED"
+  if [[ "$(command git config --get oh-my-zsh.hide-dirty 2> /dev/null)" != "1" ]]; then
+    local flags index
+    flags=('--porcelain')
+    if [[ "$DISABLE_UNTRACKED_FILES_DIRTY" == "true" ]]; then
+      flags+="--untracked-files=no"
     fi
-    if $(echo "$_INDEX" | command grep -q '^.[MTD] '); then
-      _STATUS="$_STATUS$ZSH_THEME_GIT_PROMPT_UNSTAGED"
+    index=$(command git status $flags 2> /dev/null)
+    if [[ -n "$index" ]]; then
+      if $(echo "$index" | command grep -q '^[AMRD]. '); then
+        gstatus="$gstatus$ZSH_THEME_GIT_PROMPT_STAGED"
+      fi
+      if $(echo "$index" | command grep -q '^.[MTD] '); then
+        gstatus="$gstatus$ZSH_THEME_GIT_PROMPT_UNSTAGED"
+      fi
+      if $(echo "$index" | command grep -q -E '^\?\? '); then
+        gstatus="$gstatus$ZSH_THEME_GIT_PROMPT_UNTRACKED"
+      fi
+      if $(echo "$index" | command grep -q '^UU '); then
+        gstatus="$gstatus$ZSH_THEME_GIT_PROMPT_UNMERGED"
+      fi
+    else
+      gstatus="$gstatus$ZSH_THEME_GIT_PROMPT_CLEAN"
     fi
-    if $(echo "$_INDEX" | command grep -q -E '^\?\? '); then
-      _STATUS="$_STATUS$ZSH_THEME_GIT_PROMPT_UNTRACKED"
-    fi
-    if $(echo "$_INDEX" | command grep -q '^UU '); then
-      _STATUS="$_STATUS$ZSH_THEME_GIT_PROMPT_UNMERGED"
-    fi
-  else
-    _STATUS="$_STATUS$ZSH_THEME_GIT_PROMPT_CLEAN"
   fi
 
   # check status of local repository
-  _INDEX=$(command git status --porcelain -b 2> /dev/null)
-  if $(echo "$_INDEX" | command grep -q '^## .*ahead'); then
-    _STATUS="$_STATUS$ZSH_THEME_GIT_PROMPT_AHEAD"
-  fi
-  if $(echo "$_INDEX" | command grep -q '^## .*behind'); then
-    _STATUS="$_STATUS$ZSH_THEME_GIT_PROMPT_BEHIND"
+  local ref
+  ref=$(command git rev-parse --abbrev-ref HEAD 2> /dev/null)
+  index=$(command git for-each-ref --format="%(upstream:track)" refs/heads/${ref} 2> /dev/null)
+  if $(echo "$index" | command grep -q 'ahead'); then
```

Comment:
The same as above. Or you can also use `if [[ ]]` here considering there are only two branches.
```suggestion
  if [[ "$index" == *"ahead"* ]]; then
```